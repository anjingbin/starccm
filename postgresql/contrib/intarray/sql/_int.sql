--
-- first, define the datatype.  Turn off echoing so that expected file
-- does not depend on contents of seg.sql.
--
\set ECHO none
\i _int.sql
\set ECHO all

--test query_int
select '1'::query_int;
select ' 1'::query_int;
select '1 '::query_int;
select ' 1 '::query_int;
select ' ! 1 '::query_int;
select '!1'::query_int;
select '1|2'::query_int;
select '1|!2'::query_int;
select '!1|2'::query_int;
select '!1|!2'::query_int;
select '!(!1|!2)'::query_int;
select '!(!1|2)'::query_int;
select '!(1|!2)'::query_int;
select '!(1|2)'::query_int;
select '1&2'::query_int;
select '!1&2'::query_int;
select '1&!2'::query_int;
select '!1&!2'::query_int;
select '(1&2)'::query_int;
select '1&(2)'::query_int;
select '!(1)&2'::query_int;
select '!(1&2)'::query_int;
select '1|2&3'::query_int;
select '1|(2&3)'::query_int;
select '(1|2)&3'::query_int;
select '1|2&!3'::query_int;
select '1|!2&3'::query_int;
select '!1|2&3'::query_int;
select '!1|(2&3)'::query_int;
select '!(1|2)&3'::query_int;
select '(!1|2)&3'::query_int;
select '1|(2|(4|(5|6)))'::query_int;
select '1|2|4|5|6'::query_int;
select '1&(2&(4&(5&6)))'::query_int;
select '1&2&4&5&6'::query_int;
select '1&(2&(4&(5|6)))'::query_int;
select '1&(2&(4&(5|!6)))'::query_int;


CREATE TABLE test__int( a int[] );

\copy test__int from 'data/test__int.data'

SELECT count(*) from test__int WHERE a && '{23,50}';
SELECT count(*) from test__int WHERE a @@ '23|50';
SELECT count(*) from test__int WHERE a @ '{23,50}';
SELECT count(*) from test__int WHERE a @@ '23&50';
SELECT count(*) from test__int WHERE a @ '{20,23}';
SELECT count(*) from test__int WHERE a @@ '50&68';
SELECT count(*) from test__int WHERE a @ '{20,23}' or a @ '{50,68}';
SELECT count(*) from test__int WHERE a @@ '(20&23)|(50&68)';

CREATE INDEX text_idx on test__int using gist ( a gist__int_ops );

SELECT count(*) from test__int WHERE a && '{23,50}';
SELECT count(*) from test__int WHERE a @@ '23|50';
SELECT count(*) from test__int WHERE a @ '{23,50}';
SELECT count(*) from test__int WHERE a @@ '23&50';
SELECT count(*) from test__int WHERE a @ '{20,23}';
SELECT count(*) from test__int WHERE a @@ '50&68';
SELECT count(*) from test__int WHERE a @ '{20,23}' or a @ '{50,68}';
SELECT count(*) from test__int WHERE a @@ '(20&23)|(50&68)';

drop index text_idx;
CREATE INDEX text_idx on test__int using gist ( a gist__intbig_ops );

SELECT count(*) from test__int WHERE a && '{23,50}';
SELECT count(*) from test__int WHERE a @@ '23|50';
SELECT count(*) from test__int WHERE a @ '{23,50}';
SELECT count(*) from test__int WHERE a @@ '23&50';
SELECT count(*) from test__int WHERE a @ '{20,23}';
SELECT count(*) from test__int WHERE a @@ '50&68';
SELECT count(*) from test__int WHERE a @ '{20,23}' or a @ '{50,68}';
SELECT count(*) from test__int WHERE a @@ '(20&23)|(50&68)';

