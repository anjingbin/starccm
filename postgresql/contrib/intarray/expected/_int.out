--
-- first, define the datatype.  Turn off echoing so that expected file
-- does not depend on contents of seg.sql.
--
\set ECHO none
--test query_int
select '1'::query_int;
 query_int 
-----------
 1
(1 row)

select ' 1'::query_int;
 query_int 
-----------
 1
(1 row)

select '1 '::query_int;
 query_int 
-----------
 1
(1 row)

select ' 1 '::query_int;
 query_int 
-----------
 1
(1 row)

select ' ! 1 '::query_int;
 query_int 
-----------
 !1
(1 row)

select '!1'::query_int;
 query_int 
-----------
 !1
(1 row)

select '1|2'::query_int;
 query_int 
-----------
 1 | 2
(1 row)

select '1|!2'::query_int;
 query_int 
-----------
 1 | !2
(1 row)

select '!1|2'::query_int;
 query_int 
-----------
 !1 | 2
(1 row)

select '!1|!2'::query_int;
 query_int 
-----------
 !1 | !2
(1 row)

select '!(!1|!2)'::query_int;
  query_int   
--------------
 !( !1 | !2 )
(1 row)

select '!(!1|2)'::query_int;
  query_int  
-------------
 !( !1 | 2 )
(1 row)

select '!(1|!2)'::query_int;
  query_int  
-------------
 !( 1 | !2 )
(1 row)

select '!(1|2)'::query_int;
 query_int  
------------
 !( 1 | 2 )
(1 row)

select '1&2'::query_int;
 query_int 
-----------
 1 & 2
(1 row)

select '!1&2'::query_int;
 query_int 
-----------
 !1 & 2
(1 row)

select '1&!2'::query_int;
 query_int 
-----------
 1 & !2
(1 row)

select '!1&!2'::query_int;
 query_int 
-----------
 !1 & !2
(1 row)

select '(1&2)'::query_int;
 query_int 
-----------
 1 & 2
(1 row)

select '1&(2)'::query_int;
 query_int 
-----------
 1 & 2
(1 row)

select '!(1)&2'::query_int;
 query_int 
-----------
 !1 & 2
(1 row)

select '!(1&2)'::query_int;
 query_int  
------------
 !( 1 & 2 )
(1 row)

select '1|2&3'::query_int;
 query_int 
-----------
 1 | 2 & 3
(1 row)

select '1|(2&3)'::query_int;
 query_int 
-----------
 1 | 2 & 3
(1 row)

select '(1|2)&3'::query_int;
   query_int   
---------------
 ( 1 | 2 ) & 3
(1 row)

select '1|2&!3'::query_int;
 query_int  
------------
 1 | 2 & !3
(1 row)

select '1|!2&3'::query_int;
 query_int  
------------
 1 | !2 & 3
(1 row)

select '!1|2&3'::query_int;
 query_int  
------------
 !1 | 2 & 3
(1 row)

select '!1|(2&3)'::query_int;
 query_int  
------------
 !1 | 2 & 3
(1 row)

select '!(1|2)&3'::query_int;
   query_int    
----------------
 !( 1 | 2 ) & 3
(1 row)

select '(!1|2)&3'::query_int;
   query_int    
----------------
 ( !1 | 2 ) & 3
(1 row)

select '1|(2|(4|(5|6)))'::query_int;
           query_int           
-------------------------------
 1 | ( 2 | ( 4 | ( 5 | 6 ) ) )
(1 row)

select '1|2|4|5|6'::query_int;
           query_int           
-------------------------------
 ( ( ( 1 | 2 ) | 4 ) | 5 ) | 6
(1 row)

select '1&(2&(4&(5&6)))'::query_int;
     query_int     
-------------------
 1 & 2 & 4 & 5 & 6
(1 row)

select '1&2&4&5&6'::query_int;
     query_int     
-------------------
 1 & 2 & 4 & 5 & 6
(1 row)

select '1&(2&(4&(5|6)))'::query_int;
       query_int       
-----------------------
 1 & 2 & 4 & ( 5 | 6 )
(1 row)

select '1&(2&(4&(5|!6)))'::query_int;
       query_int        
------------------------
 1 & 2 & 4 & ( 5 | !6 )
(1 row)

CREATE TABLE test__int( a int[] );
\copy test__int from 'data/test__int.data'
SELECT count(*) from test__int WHERE a && '{23,50}';
 count 
-------
   403
(1 row)

SELECT count(*) from test__int WHERE a @@ '23|50';
 count 
-------
   403
(1 row)

SELECT count(*) from test__int WHERE a @ '{23,50}';
 count 
-------
    12
(1 row)

SELECT count(*) from test__int WHERE a @@ '23&50';
 count 
-------
    12
(1 row)

SELECT count(*) from test__int WHERE a @ '{20,23}';
 count 
-------
    12
(1 row)

SELECT count(*) from test__int WHERE a @@ '50&68';
 count 
-------
     9
(1 row)

SELECT count(*) from test__int WHERE a @ '{20,23}' or a @ '{50,68}';
 count 
-------
    21
(1 row)

SELECT count(*) from test__int WHERE a @@ '(20&23)|(50&68)';
 count 
-------
    21
(1 row)

CREATE INDEX text_idx on test__int using gist ( a gist__int_ops );
SELECT count(*) from test__int WHERE a && '{23,50}';
 count 
-------
   403
(1 row)

SELECT count(*) from test__int WHERE a @@ '23|50';
 count 
-------
   403
(1 row)

SELECT count(*) from test__int WHERE a @ '{23,50}';
 count 
-------
    12
(1 row)

SELECT count(*) from test__int WHERE a @@ '23&50';
 count 
-------
    12
(1 row)

SELECT count(*) from test__int WHERE a @ '{20,23}';
 count 
-------
    12
(1 row)

SELECT count(*) from test__int WHERE a @@ '50&68';
 count 
-------
     9
(1 row)

SELECT count(*) from test__int WHERE a @ '{20,23}' or a @ '{50,68}';
 count 
-------
    21
(1 row)

SELECT count(*) from test__int WHERE a @@ '(20&23)|(50&68)';
 count 
-------
    21
(1 row)

drop index text_idx;
CREATE INDEX text_idx on test__int using gist ( a gist__intbig_ops );
SELECT count(*) from test__int WHERE a && '{23,50}';
 count 
-------
   403
(1 row)

SELECT count(*) from test__int WHERE a @@ '23|50';
 count 
-------
   403
(1 row)

SELECT count(*) from test__int WHERE a @ '{23,50}';
 count 
-------
    12
(1 row)

SELECT count(*) from test__int WHERE a @@ '23&50';
 count 
-------
    12
(1 row)

SELECT count(*) from test__int WHERE a @ '{20,23}';
 count 
-------
    12
(1 row)

SELECT count(*) from test__int WHERE a @@ '50&68';
 count 
-------
     9
(1 row)

SELECT count(*) from test__int WHERE a @ '{20,23}' or a @ '{50,68}';
 count 
-------
    21
(1 row)

SELECT count(*) from test__int WHERE a @@ '(20&23)|(50&68)';
 count 
-------
    21
(1 row)

